src/contexts/authContext.js

```diff
@@ -1,82 +1,74 @@
-// import { useNavigate } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
 
 const AuthContext = createContext();
 
-const AuthProvider = ({children}) => {
-
-    const [state, dispatch] = useReducer( authReducer, { firstName: "", token: "" });
+const initialState = {
+    userName: JSON.parse(localStorage.getItem("jwtToken"))?.userName || "",
+    token: JSON.parse(localStorage.getItem("jwtToken"))?.token || "",
+    isAuth: JSON.parse(localStorage.getItem("jwtToken"))?.isAuth || false,
+}
 
-    // const navigate = useNavigate();
+const AuthProvider = ({children}) => {
 
-    
+    const [error, setError] = useState("");
+    const [errorState, setErrorState] = useState(false);
+    const [state, dispatch] = useReducer( authReducer, initialState);
+    const navigate = useNavigate();
 
     const signup = async ({firstName, lastName, email, password}) => {
-
         try {
-            const response = await axios.post("/api/auth/signup", {
-                firstName: firstName,
-                lastName: lastName,
-                email: email,
-                password: password,
-            });
+            const response = await axios.post("/api/auth/signup", {firstName, lastName, email, password});
 
             if(response.status === 201) {
-                // console.log("response", response);
-                const {createdUser} = response.data.createdUser;
+                const fName = response.data.createdUser.firstName;
                 const encodedToken = response.data.encodedToken;
-                localStorage.setItem("jwtToken", JSON.stringify({userInfo: createdUser, token: encodedToken}));
-                dispatch({type: "LOGIN", payload: {userInfo: createdUser, token: encodedToken}});
-                // navigate("/");
+
+                localStorage.setItem("jwtToken", JSON.stringify({userName: fName, token: encodedToken, isAuth: true}));
+                dispatch({type: "LOGIN", payload: {name: fName, token: encodedToken, isAuth: true},});
+                navigate("/");
             }
         } catch (error) {
-            
-            // setPwdError(error);
-            // console.log(pwdError);
-            console.log("error recieved");
+            setError(error.response.data.errors);
+            setErrorState(true);
+            setTimeout(() => {
+                setErrorState(false);
+            }, 3000);
         }
     }
 
-    // *** LOGIN *** //
-
-    const login = async ({firstName, lastName, email, password}) => {
-        
+    const login = async ({email, password}) => {
         try{
-
-            const response = await axios.post("/api/auth/signup", {
-                firstName: firstName,
-                lastName: lastName,
-                email: email,
-                password: password,
-            });
-
+            const response = await axios.post("/api/auth/logi", {email, password});
+            
```

Comment:
Bug: endpoint has spelling mistake."login*"

Reply:
I think all the new changes are not updated in the code after last commit.

Reply:
Removed all the comments and bugs as well in the new one.